test/parallel/test-runner-reporters.js

```diff
@@ -94,4 +94,26 @@ describe('node:test reporters', { concurrency: true }, () => {
       assert.strictEqual(stdout.slice(0, filename.length + 2), `${filename} {`);
     });
   });
+
+  it('should support a custom reporter from node_modules', async () => {
+    const child = spawnSync(process.execPath,
+                            ['--test', '--test-reporter', 'reporter-cjs', 'reporters.js'],
+                            { cwd: fixtures.path('test-runner') });
+    assert.strictEqual(child.stderr.toString(), '');
+    const stdout = child.stdout.toString();
+    assert.match(stdout, /{"test:start":5,"test:pass":2,"test:fail":3,"test:plan":3,"test:diagnostic":\d+}$/);
+    const label = 'package: reporter-cjs';
+    assert.strictEqual(stdout.slice(0, label.length + 1), `${label}{`);
```

Comment:
```suggestion
    const stdout = ;
    assert.match(
      child.stdout.toString(),
      /^package: reporter-cjs{"test:start":5,"test:pass":2,"test:fail":3,"test:plan":3,"test:diagnostic":\d+}$/,
    );
```

Reply:
What’s the difference? That you do all the comparisons in one `assert.match` rather than two `assert` calls?

I was copying what just landed in #45930:

https://github.com/nodejs/node/blob/71951a0e86da9253d7c422fa2520ee9143e557fa/test/parallel/test-runner-reporters.js#L93-L94

If you want to refactor for style, could that happen in a follow-up? All of this test rewriting is unrelated to my goals in this PR, and what I was trying to avoid by having unrelated tests added to this.

Reply:
See https://github.com/nodejs/node/pull/45930#discussion_r1054294280.